### ISSUE TITLE
Fedora known issues

### ISSUE THREAD

**Issue by dkosovic** [nm-l2tp/network-manager-l2tp#13]
## Prerequisite L2TP kernel module issue

Due to Fedora Bugzilla bug [#1284228](https://bugzilla.redhat.com/show_bug.cgi?id=1284228), before installing _xl2tpd_ or _NetworkManager-l2tp_, it is recommended to explicitly install _kernel-modules-extra_  :

``` sh
sudo dnf install kernel-modules-extra 
```

otherwise _kernel-debug-modules-extra_ will get installed instead.
## Installing NetworkManager-l2tp

The following installation instructions are for Fedora 23 and later.

Issue the following to install NetworkManager-l2tp and its GNOME components :

``` sh
dnf install NetworkManager-l2tp-gnome
```

For KDE desktops environments:

``` sh
dnf install NetworkManager-l2tp
```

as NetworkManager-l2tp-gnome is not required for KDE.

**Comment by kusznir** [comment]
I'm having difficulty with this plugin on Fedora 23.  It insists on installing and using libreswan, which fails to rekey my vpn when the rekey interval hits.  Others online say I should be using strongswan, and it is installed, and I gather that this plugin can use it, but I'm at a loss on how to make it....

**Comment by dkosovic** [comment]
The following should be enough to use strongswan:
```
dnf install strongswan
rpm -e --nodeps libreswan
killall -TERM nm-l2tp-service
```

**Comment by kusznir** [comment]
Further work found the core problem.  Switching to StrongSwan is not likely to fix it.  NetworkManager deletes the secrets file shortly after establishing the connection.  Unfortunately, this means that when the ipsec connection comes up for rekey, it fails because it can't authenticate with the pre-shared key anymore.  I was successfully able to work around my problem by manually creating a .secrets file with my PSK in it.  However, I believe this is a bug in the NM module; it should not remove the .secrets file it puts in until after the link is down.

**Comment by dkosovic** [comment]
Main issue is than none of the major Linux distributions other than Fedora (and derivates) do the following in the `/etc/ipsec.secrets` file:
   include /etc/ipsec.d/*.secrets

So ideally would like to have the following file during the lifetime of the VPN connection, much like what NetworkManager-libreswan does:
  /etc/ipsec.d/ipsec-conn-name.secrets

What is happening now with NetworkManager-l2tp is that `/etc/ipsec.secrets` is replaced briefly and the original restored. The reason `/etc/ipsec.secrets` is restored early is because the original file may never be restored otherwise for example if the link goes down ungracefully with a SIGKILL, etc.

I've got to think about, might have to have a configure script that checks if /etc/ipsec.secrets contains the the include line.

**Comment by kusznir** [comment]
I fixed it for me by adding the /etc/ipsec.d/connection.secrets file.

Perhaps the NM plugin will check /etc/ipsec.secrets file, and if it doesn't
have the include line, add it (which it can leave), then create a .secrets
file.  Could do connection.secrets, or connection-pid.secrets (and clean up
pids not running whenever it writes out a file).

**Comment by dkosovic** [comment]
NetworkManager-l2tp versions 1.0.6 and 1.2.6 now use `/etc/ipsec.d/nm-l2tp-ipsec-UUID.secrets` as described here :

https://github.com/nm-l2tp/network-manager-l2tp#run-time-generated-files

**Comment by dkosovic** [comment]
Closing this issue as there is no longer a requirement to explicitly install _kernel-modules-extra_  as it has been fixed in [bug# 1192189]( https://bugzilla.redhat.com/show_bug.cgi?id=1192189)

Also packaged install instructions are now listed here:
* https://github.com/nm-l2tp/network-manager-l2tp/wiki/Prebuilt-Packages
